systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -721,30 +742,54 @@ void testCertRenewalInMaintenanceWindow(ExtensionContext extensionContext) {
 
         Map<String, String> kafkaPods = PodUtils.podSnapshot(namespaceName, kafkaSelector);
 
-        LOGGER.info("Annotate secret {} with secret force-renew annotation", secretName);
-        Secret secret = new SecretBuilder(kubeClient(namespaceName).getSecret(namespaceName, secretName))
-            .editMetadata()
-                .addToAnnotations(Ca.ANNO_STRIMZI_IO_FORCE_RENEW, "true")
-            .endMetadata().build();
-        kubeClient(namespaceName).patchSecret(namespaceName, secretName, secret);
-
-        LOGGER.info("Wait until maintenance windows starts");
-        LocalDateTime finalMaintenanceWindowStart = maintenanceWindowStart;
-        TestUtils.waitFor("maintenance window start",
-            Constants.GLOBAL_POLL_INTERVAL, Duration.ofMinutes(maintenanceWindowDuration).toMillis() - 10000,
-            () -> LocalDateTime.now().isAfter(finalMaintenanceWindowStart));
+        CertificateAuthority newCAValidity = new CertificateAuthority();
+        newCAValidity.setRenewalDays(150);
+        newCAValidity.setValidityDays(200);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(
+            testStorage.getClusterName(),
+            k -> {
+                k.getSpec().setClusterCa(newCAValidity);
+                k.getSpec().setClientsCa(newCAValidity);
+            },
+            testStorage.getNamespaceName()
+        );
 
-        LOGGER.info("Maintenance window starts");
+        TestUtils.waitFor("Wait for reconciliation to happen and make sure neither Kafka nor CA certs were renewed (as we're outside of maintenance time window).",
+            Constants.GLOBAL_POLL_INTERVAL_MEDIUM, Constants.SAFETY_RECONCILIATION_INTERVAL,
+            () -> {
+                Secret secretCaCluster = kubeClient(namespaceName).getSecret(namespaceName, clusterSecretName);
+                Secret secretCaClients = kubeClient(namespaceName).getSecret(namespaceName, clientsSecretName);
+                return secretCaCluster.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION).equals("0")
+                    && secretCaClients.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION).equals("0")
+                    && kafkaPods.equals(PodUtils.podSnapshot(namespaceName, kafkaSelector));
+            }
+        );
```

Comment:
Wouldn't be better to add this to some `SecretUtils`?

Reply:
I am not sure, that's only one usage. Do anticipate more of such use cases?

Reply:
Fixed after all, looks better :-)